gateway.go

```diff
@@ -159,29 +158,17 @@ func (gw *Gateway) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Ms
 	log.Debugf("computed response addresses %v", addrs)
 
 	// Fall through if no host matches
-	if len(addrs) == 0 && gw.Fall.Through(qname) {
+	if len(addrs["A"]) == 0 && len(addrs["AAAA"]) == 0 && len(addrs["TXT"]) == 0 && gw.Fall.Through(qname) {
```

Comment:
Why?

Reply:
Because I split the different record types. To be able to generalize with types that are not ip addresses

Reply:
You could have left them as they are, and introduce another array for them instead? I don't understand why you decided that it needed to be generalised at this moment.